core/src/main/java/org/infinispan/interceptors/locking/AbstractTxLockingInterceptor.java

```diff
@@ -109,15 +100,55 @@ protected final Object invokeNextAndCommitIf1Pc(TxInvocationContext ctx, Prepare
     * locks to be released. The backup lock will be released either by a commit/rollback/unlock command or by
     * the originator leaving the cluster (if recovery is disabled).
     */
-   protected final void lockAndRegisterBackupLock(TxInvocationContext ctx, Object key, long lockTimeout, boolean skipLocking) throws InterruptedException {
-      //with DeltaCompositeKey, the locks should be acquired in the owner of the delta aware key.
-      Object keyToCheck = key instanceof DeltaCompositeKey ?
-            ((DeltaCompositeKey) key).getDeltaAwareValueKey() :
-            key;
-      if (cdl.localNodeIsPrimaryOwner(keyToCheck)) {
-         lockKeyAndCheckOwnership(ctx, key, lockTimeout, skipLocking);
-      } else if (cdl.localNodeIsOwner(keyToCheck)) {
-         ctx.getCacheTransaction().addBackupLockForKey(key);
+   protected final void lockAndRegisterBackupLock(TxInvocationContext<?> ctx, Object key, long lockTimeout, Action action)
```

Comment:
I know this method was already named this, but should it not be an or instead of and?